src/Helper/PaymentHelper.php

```diff
@@ -114,33 +143,67 @@ public function validatePayment($endpoint): bool {
     );
     $result = $this->responseToObject($response);
 
-    $reservedAmount = $result->payment->summary->reservedAmount ?? NULL;
-    $chargedAmount = $result->payment->summary->chargedAmount ?? NULL;
+    $amountToPay = floatval($this->privateTempStore->get(NetsEasyPaymentElement::AMOUNT_TO_PAY) * 100);
+    $reservedAmount = floatval($result->payment->summary->reservedAmount);
+    $chargedAmount = floatval($result->payment->summary->chargedAmount);
+
+    if ($amountToPay !== $reservedAmount) {
+      // Reserved amount mismatch.
+      $formState->setError(
+        $element,
+        'Reserved amount mismatch'
+      );
+      return;
+    }
 
     if ($reservedAmount && !$chargedAmount) {
       // Payment is reserved, but not yet charged.
-      $paymentCharged = $this->chargePayment($endpoint, $reservedAmount);
-      return $paymentCharged;
-    }
+      $paymentChargeId = $this->chargePayment($paymentEndpoint, $reservedAmount);
+      if (!$paymentChargeId) {
+        $formState->setError(
+          $element,
+          'Payment was not charged'
+        );
+        return;
+      }
 
-    if (!$reservedAmount && !$chargedAmount) {
-      // Reservation was not made.
-      return FALSE;
+      $chargeEndpoint = $this->getChargeEndpoint() . $paymentChargeId;
+      $response = $this->httpClient->request(
+      'GET',
+      $chargeEndpoint,
+      [
+        'headers' => [
+          'Content-Type' => 'application/json',
```

Comment:
There's no reason to send a `content-type` header on `GET` requests (cf. https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type).